typescript/src/main.ts

```diff
@@ -178,35 +178,72 @@ function reset() {
   calculate();
 }
 
+function cleanData(data: string) {
+  data = data.replace(/(\r\n|\n|\r)/gm, "").replace(/(\s\s)/gm, " ");
+  data = data.replace(/"/g, '""');
+  data = data.replace(/\u2206/g, "Delta ");
+  return data;
+}
+function convertToPlainText(html: string) {
+  // Create a new div element
+  const tempDivElement = document.createElement("div");
+  // Set the HTML content with the given value
+  tempDivElement.innerHTML = html;
+  // Retrieve the text property of the element
+  return tempDivElement.textContent || tempDivElement.innerText || "";
+}
+
 // Quick and simple export target #tableId into a csv
 function downloadTableAsCsv(tableId: string, separator = ",") {
-  // Select rows from tableId
-  const rows = document.querySelectorAll("table#" + tableId + " tr");
+  const table = <HTMLTableElement>document.getElementById(tableId);
+  const rows = table.rows;
   // Construct csv
   const csv = [];
   for (let i = 0; i < rows.length; i++) {
     const row = [];
-    const cols = <NodeListOf<HTMLTableCellElement>>(
-      rows[i].querySelectorAll("td, th")
-    );
+    const cols = rows[i].cells;
     for (let j = 0; j < cols.length; j++) {
       // Clean innertext to remove multiple spaces and jumpline (break csv)
       let data;
-      const inputs = cols[j].getElementsByTagName("input");
+
+      const unfilteredInputs = cols[j].getElementsByTagName("input");
+      const inputs = [];
+      if (unfilteredInputs.length > 0) {
+        for (let l = 0; l < unfilteredInputs.length; l++) {
+          const input = unfilteredInputs[l];
+          if (input.type == "number") {
+            inputs.push(input);
+          }
+        }
+      }
+
```

Comment:
I could feel there must be a better way of doing this just didn't know how! Learnt something new today!

Reply:
<img width="901" alt="Screenshot 2022-01-17 at 13 35 04" src="https://user-images.githubusercontent.com/73976500/149778581-48b42a8f-f1b6-4e79-b2f9-c607443d01be.png">

Reply:
The line I gave was probably JS from testing on the console rather than typescript - sorry!

Try something like:

`Array.from(document.getElementsByTagName('input')).filter(e => e.type == 'checkbox')`